src/Core/Services/MetadataProviders/MetadataProviderFactory.cs

```diff
@@ -62,7 +62,10 @@ private void ConfigureMetadataProviders()
 
         public void OnConfigChanged(object? sender, HotReloadEventArgs args)
         {
+            _metadataProviders = new Dictionary<string, ISqlMetadataProvider>();
             ConfigureMetadataProviders();
+            // Blocks the current thread until initialization is finished.
+            this.InitializeAsync().GetAwaiter().GetResult();
```

Comment:
#2417 handles this in a different way. Can you and @abhishekkumams identify which method you want to ultimately use?

Reply:
they are the same, but this way will give us the internal exceptions that come from initializeAsync, instead of wrapping them in an aggregate exception.